src/app/tx/[tx]/MessageDisplays.tsx

```diff
@@ -50,26 +50,40 @@ const getDataFromL1ToL2Message = (
   };
 };
 
+type MessageStatusDisplayWithAutoRedeem = MessageStatusDisplay & {
+  autoRedeemHash: string | undefined;
+};
 async function getData({
   retryables: l1ToL2Messages,
   retryablesClassic: l1ToL2MessagesClassic,
   deposits,
-}: L1ToL2MessagesAndDepositMessages) {
-  const l1ToL2MessagesPromises = l1ToL2Messages.map((l1ToL2Message) =>
-    l1ToL2MessageToStatusDisplay(l1ToL2Message, false),
-  );
+}: L1ToL2MessagesAndDepositMessages): Promise<{
+  messagesDisplays: MessageStatusDisplayWithAutoRedeem[];
+}> {
+  const l1ToL2MessagesPromises = l1ToL2Messages.map(async (l1ToL2Message) => {
+    return {
+      ...(await l1ToL2MessageToStatusDisplay(l1ToL2Message, false)),
```

Comment:
The function `getData` was quite clean and readable earlier.

I would suggest moving the autoRedeem business logic to `l1ToL2MessageToStatusDisplay` only.
In `l1ToL2MessageToStatusDisplay` we can conditionally call auto redeem getter if the flag is supplied by the user.

The new contract for `l1ToL2MessageToStatusDisplay` can be
`l1ToL2MessageToStatusDisplay(params:{l1ToL2Message, isClassic:boolean, withAutoRedeem:boolean})`

Reply:
Make sense, I moved it and simplified the logic. I don't think we need a flag, as it's only used here, so fetching by default is fine for now